### PR TITLE
feature/api doctor net core

### PR DESCRIPTION
**Pull request by baywet**
Upgrades the build sequence to dotnet 5 for API doctor as part of typewriter dotnet 5 migration work.

- migrates scripts to dotnet core
- upgrades build sequence for API doctor to dotnet 5

### CODE REVIEW

**Comment by baywet** [comment]
FYI the updated pipeline "successfully failed" as they were no changes to commit [here](https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=33938&view=results)

**Comment by baywet** [comment]
also migrated from a windows base to ubuntu for performance. [Here](https://microsoftgraph.visualstudio.com/Graph%20Developer%20Experiences/_build/results?buildId=33969&view=results) is the "successfully failing" build.

image

1. dotnet classic on windows
1. dotnet core on windows
1. dotnet core on ubuntu

=> 50% performance improvement

**Comment by zengin** [comment]
woot! Linux all the things!

Most likely the improvement is because of fast file system in Linux as this pipeline uses file system to pass snippets between tools.

File: pipelines/snippets.yml [hunk 1]
```diff
@@ -52,94 +52,78 @@ steps:
...
 
-      $apidoctorPath = (Get-ChildItem $env:Build_SourcesDirectory\apidoctor\ApiDoctor.Console\bin\Release apidoc.exe -Recurse).FullName
-      Write-Host "Path to apidoctor tool: $apidoctorPath"
+- pwsh: |
+    # release folder can change based on .NET core version, so search recursively in bin folder
+    $snippetGeneratorPath = (Get-ChildItem $env:BUILD_SOURCESDIRECTORY/microsoft-graph-explorer-api/CodeSnippetsReflection.App/bin/Release *App -Recurse).FullName
+    Write-Host "Path to snippet generator tool: $snippetGeneratorPath"
 
-      & $apidoctorPath generate-snippets --ignore-warnings --path . --snippet-generator-path $snippetGeneratorPath --lang $(snippetLanguages) --git-path "C:\Program Files\Git\bin\git.exe"
+    $apidoctorPath = (Get-ChildItem $env:BUILD_SOURCESDIRECTORY/apidoctor/ApiDoctor.Console/bin/Release/net5.0 apidoc -Recurse).FullName
```
Comment (zengin) [inline pipelines/snippets.yml:103]: Given that we have `-Recurse`, I assume we can safely remove `net5.0` to make it work for future dotnet releases without a code change.

```suggestion
    $apidoctorPath = (Get-ChildItem $env:BUILD_SOURCESDIRECTORY/apidoctor/ApiDoctor.Console/bin/Release apidoc -Recurse).FullName
```

File: pipelines/snippets.yml [hunk 2]
```diff
@@ -52,94 +52,78 @@ steps:
...
+    Write-Host "The new branch name will be: $env:BRANCHNAME"
+    git checkout -B $env:BRANCHNAME | Write-Host
   displayName: 'Git: branch from master named with the build id: $(Build.BuildId)'
-  inputs:
-    targetType: inline
-    workingDirectory: '$(Build.SourcesDirectory)/microsoft-graph-docs'
-    script: |
-      Write-Host "The new branch name will be: $env:branchName"
-      git checkout -B $env:branchName | Write-Host
+  workingDirectory: '$(Build.SourcesDirectory)/microsoft-graph-docs'
```
Comment (zengin) [inline pipelines/snippets.yml:65]: I have recently realized that this works too, I think it is easier to read this way.
```suggestion
  workingDirectory: 'microsoft-graph-docs'
```

File: pipelines/snippets.yml [hunk 3]
```diff
@@ -52,94 +52,78 @@ steps:
...
 
-- task: PowerShell@2
+- pwsh: |
+    git config user.email "GraphTooling@service.microsoft.com"
+    git config user.name "Microsoft Graph DevX Tooling"
   displayName: 'Git: set user config'
+  workingDirectory: '$(Build.SourcesDirectory)/microsoft-graph-docs'
```
Comment (zengin) [inline pipelines/snippets.yml:71]:
```suggestion
  workingDirectory: 'microsoft-graph-docs'
```

File: pipelines/snippets.yml [hunk 4]
```diff
@@ -52,94 +52,78 @@ steps:
...
-        git commit -m "Update generated files with build $env:Build_BuildId [skip ci]" | Write-Host
-      }
-      else
-      {
-        git commit -m "Update generated files with build $env:Build_BuildId" | Write-Host
-      }
-      Write-Host "Added and commited generated files." -ForegroundColor Green
-
-- task: PowerShell@2
+  workingDirectory: '$(Build.SourcesDirectory)/microsoft-graph-docs'
```
Comment (zengin) [inline pipelines/snippets.yml:123]:
```suggestion
  workingDirectory: 'microsoft-graph-docs'
```

File: pipelines/snippets.yml [hunk 5]
```diff
@@ -52,94 +52,78 @@ steps:
...
+    Write-Host "Pushed the results of the build to the $env:BRANCHNAME branch." -ForegroundColor Green
   displayName: 'Git: push updates'
-  inputs:
-    targetType: inline
-    workingDirectory: '$(Build.SourcesDirectory)/microsoft-graph-docs'
-    script: |
-      git push --set-upstream origin $env:branchName | Write-Host
-      Write-Host "Pushed the results of the build to the $env:branchName branch." -ForegroundColor Green
-  enabled: true
+  workingDirectory: '$(Build.SourcesDirectory)/microsoft-graph-docs'
```
Comment (zengin) [inline pipelines/snippets.yml:129]:
```suggestion
  workingDirectory: 'microsoft-graph-docs'
```

**Review by zengin — APPROVED** [review]
Moving away from .NET framework for ApiDoctor will give us a lot of perf benefits. Thanks for doing this.

**Review by zengin — APPROVED** [review]

**Review by MIchaelMainer — APPROVED** [review]

**Pull request merged**